### PR TITLE
Adds camera modes + photographer trait uses

### PR DESCRIPTION
**Pull request by Hopekz**
Adds modes to the camera. 

**What this means:** 
You can set your camera to shoot in standard mode or description mode by using the camera in your hand. 

In standard mode, the camera will not prompt you if you want to customize the photo and in description mode, it gives you that annoying popup asking you if you want to customize your photo.

**Edit 2:**
Added a better examine (now tells you that the camera is empty when there are 0 pictures left)
You now have to be close to the camera to look inside of it to see the amount of pictures left for added camera immersion!

**Edit 3:**
Added more use for the Photography trait.

When you have the photography trait you get the following information when you examine:

the exact number of photos left in the camera
the lens setting of the camera
the mode the camera is in

**EDIT 4:**
I managed to recreate the bug after multiple playtests so I am no longer claiming it is fixed.

#### Changelog

:cl:  Hopek
rscadd: The camera has modes. You can now shoot in a standard mode where it won't give you an annoying popup box and a description mode where it prompts you.
rscadd: Expanded the usage of the photographer trait for the Camera.
/:cl:

### CODE REVIEW

**Comment by Hopekz** [comment]
Also fixes:
#6389
#8317

**Comment by Hopekz** [comment]
Update:
Added future proofing to include a case that doesn't exist yet if someone were to make a camera that doesn't have the ability to use descriptive mode (if it can't change descriptions)

**Comment by Hopekz** [comment]
Edit 3:
Added more use for the Photography trait.

When you have the photography trait you get the following information when you examine:
- the exact number of photos left in the camera
- the lens setting of the camera
- the mode the camera is in

**Comment by MegaEmpirical** [comment]
Fuck yeah.

**Comment by Hopekz** [comment]
@swissloaf stop fucking with our labels or you'll get that privilege removed . 
This is the 2nd PR where you added or removed the ready label it's for our organization

**Comment by swissloaf** [comment]
This aint approved change

**Comment by Hopekz** [comment]
@alexkar598 you may have forgotten to submit your review with suggestions on why you believe this isn't "ready".

**Comment by alexkar598** [comment]
It's not "ready" because you cant review your own code

**Comment by Hopekz** [comment]
> It's not "ready" because you cant review your own code

I was not aware the "ready" tag had any real meaning outside of organization since we need to perform a final review before merging.

In typical non self merge circumstances; I am fully capable of performing my own review as long as the review itself is not the sole reason for merge.

You have stated yourself that the "ready" tag has no further meaning outside of a comment approval.
It does not mean that the HC or maintainer that looks in to this in the future will bypass their code review as performing a code review before a merge should be mandatory (if it isn't already).

If we are expected to bypass our final review on PR's marked as "ready" we need to communicate this out better so everyone is on the same page.

File: code/modules/photography/camera/camera.dm
Comment (alexkar598) [inline code/modules/photography/camera/camera.dm]: Move this to the initial value instead

File: code/modules/photography/camera/camera.dm
Comment (alexkar598) [inline code/modules/photography/camera/camera.dm]: Cleaner this way if a map wants to have a camera with 1 picture but you can still refill to 10 or whatever
```suggestion
	var/pictures_left = 0
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]
Use defines instead of directly using strings for the mode

**Comment by Hopekz** [comment]
> Use defines instead of directly using strings for the mode

@alexkar598  done

File: code/modules/photography/camera/camera.dm
Comment (JamieD1) [inline code/modules/photography/camera/camera.dm]: Random spaces?

**Review by JamieD1 — APPROVED** [review]

File: code/modules/photography/camera/camera.dm
Comment (alexkar598) [inline code/modules/photography/camera/camera.dm]: Why the desired_ checks? Unless the user enters 0, this will always run. If typing in 0 to keep the old size is intended behaviour, it is bad behaviour

File: code/modules/photography/camera/camera.dm
Comment (alexkar598) [inline code/modules/photography/camera/camera.dm]: Duplicate and useless code in here
```suggestion
/obj/item/camera/attack_self(mob/user)
	if(can_switch_modes)
		if(camera_mode == CAMERA_STANDARD && can_customise)
			camera_mode = CAMERA_DESCRIPTION
		else if(camera_mode == CAMERA_DESCRIPTION)
			camera_mode = CAMERA_STANDARD
		to_chat(user, "<span class='notice'>You set the [src] to [camera_mode] mode.</span>")
		return

	to_chat(user, "<span class='notice'>This [src] can only be used in the [camera_mode] mode.</span>") // just in-case somone makes a camera that can only be descriptive
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]
Il have to admit im not a fan of making cameras even more useless to just about 90% of people

File: code/modules/photography/camera/camera.dm
Comment (Hopekz) [reply code/modules/photography/camera/camera.dm]: I can't argue with this logic.
Useless checks bad!

**Comment by Hopekz** [comment]
> Il have to admit im not a fan of making cameras even more useless to just about 90% of people

The only solution I can see is to change the photographer trait to "has a camera" and just make everyone have equal proficiency with the tool or removing this as a trait. Being a photographer means you get expertise in this specific area in the game.

With these trait changes, picture-taking remains unaffected but you get all the complicated information on examination similar to someone knowing how to use the tool proficiently.

In my opinion, we should expand this further to other areas of the game like guns because right now you are aware of the exact number of bullets in a nonclear magazine just by looking at it which is weird. This should be reserved for people that are more knowledgeable in these items but that is a separate PR.
Similar to this you're aware of the exact number of pictures left in a camera just by looking at it.
To me, it seems like this is something that proficiency would let you know.

**Also if you're planning to close this PR just request the edit of it being removed because the main focus of this PR is adding the camera modes so it doesn't spam the popup every time you take a picture.**

**Review by alexkar598 — APPROVED** [review]

**Pull request merged**
